haskellPackages.hledger-flow: missing linked libraries

```
$ nix-shell -p haskellPackages.hledger-flow
these derivations will be built:
  /nix/store/yv96s715lppv7x50a9wkn64x4arp04n2-hledger-flow-0.12.3.1.drv
building '/nix/store/yv96s715lppv7x50a9wkn64x4arp04n2-hledger-flow-0.12.3.1.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5.
unpacking sources
unpacking source archive /nix/store/5p7b2md4f248ccqmr73hgg6bfbgv9mmg-hledger-flow-0.12.3.1.tar.gz
source root is hledger-flow-0.12.3.1
setting SOURCE_DATE_EPOCH to timestamp 1561311906 of file hledger-flow-0.12.3.1/test/TestHelpers.hs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/setup-package.conf.d -j4 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /build/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --docdir=/nix/store/mzjyh9m950h1pr3wwvxn2znrdjri87a0-hledger-flow-0.12.3.1-doc/share/doc/hledger-flow-0.12.3.1 --with-gcc=gcc --package-db=/build/package.conf.d --ghc-option=-j4 --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --extra-lib-dirs=/nix/store/z5dab4jnb0glqvkp3iczwy85100wn6g8-ncurses-6.1-20190112/lib --extra-lib-dirs=/nix/store/phn6ipx2dxyaqcw1hvkix5409nfp5mdi-libffi-3.2.1/lib --extra-lib-dirs=/nix/store/f0wi5c6j4zl8idrd6qw3bsis7cmdy7j7-gmp-6.1.2/lib
Using Parsec parser
Configuring hledger-flow-0.12.3.1...
Dependency base >=4.7 && <5: using base-4.12.0.0
Dependency containers -any: using containers-0.6.0.1
Dependency foldl -any: using foldl-1.4.5
Dependency stm -any: using stm-2.5.0.0
Dependency text -any: using text-1.2.3.1
Dependency time -any: using time-1.8.0.2
Dependency turtle -any: using turtle-1.5.14
Dependency base >=4.7 && <5: using base-4.12.0.0
Dependency hledger-flow -any: using hledger-flow-0.12.3.1
Dependency optparse-applicative -any: using optparse-applicative-0.14.3.0
Dependency text -any: using text-1.2.3.1
Dependency turtle -any: using turtle-1.5.14
Dependency HUnit -any: using HUnit-1.6.0.0
Dependency base >=4.7 && <5: using base-4.12.0.0
Dependency containers -any: using containers-0.6.0.1
Dependency foldl -any: using foldl-1.4.5
Dependency hledger-flow -any: using hledger-flow-0.12.3.1
Dependency stm -any: using stm-2.5.0.0
Dependency text -any: using text-1.2.3.1
Dependency turtle -any: using turtle-1.5.14
Source component graph:
    component lib
    component exe:hledger-flow dependency lib
    component test:hledger-flow-test dependency lib
Configured component graph:
    component hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include base-4.12.0.0
        include containers-0.6.0.1
        include foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        include stm-2.5.0.0
        include text-1.2.3.1
        include time-1.8.0.2
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
    component hledger-flow-0.12.3.1-ILYnJMEu9ISLs6tRFyoEBS-hledger-flow
        include base-4.12.0.0
        include hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include optparse-applicative-0.14.3.0-KgUFYVq9ZZSIX6QVzTkb7A
        include text-1.2.3.1
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
    component hledger-flow-0.12.3.1-34mSVQoxqcJ9XqVJgwZ6l9-hledger-flow-test
        include HUnit-1.6.0.0-LOgQcU2eAMywe67LHWfdn
        include base-4.12.0.0
        include containers-0.6.0.1
        include foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        include hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include stm-2.5.0.0
        include text-1.2.3.1
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
Linked component graph:
    unit hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include base-4.12.0.0
        include containers-0.6.0.1
        include foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        include stm-2.5.0.0
        include text-1.2.3.1
        include time-1.8.0.2
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
        Hledger.Flow.CSVImport=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.CSVImport,Hledger.Flow.Common=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.Common,Hledger.Flow.Import.Types=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.Import.Types,Hledger.Flow.Reports=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.Reports,Hledger.Flow.RuntimeOptions=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.RuntimeOptions,Hledger.Flow.Types=hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM:Hledger.Flow.Types
    unit hledger-flow-0.12.3.1-ILYnJMEu9ISLs6tRFyoEBS-hledger-flow
        include base-4.12.0.0
        include hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include optparse-applicative-0.14.3.0-KgUFYVq9ZZSIX6QVzTkb7A
        include text-1.2.3.1
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
    unit hledger-flow-0.12.3.1-34mSVQoxqcJ9XqVJgwZ6l9-hledger-flow-test
        include HUnit-1.6.0.0-LOgQcU2eAMywe67LHWfdn
        include base-4.12.0.0
        include containers-0.6.0.1
        include foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        include hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        include stm-2.5.0.0
        include text-1.2.3.1
        include turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
Ready component graph:
    definite hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        depends base-4.12.0.0
        depends containers-0.6.0.1
        depends foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        depends stm-2.5.0.0
        depends text-1.2.3.1
        depends time-1.8.0.2
        depends turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
    definite hledger-flow-0.12.3.1-ILYnJMEu9ISLs6tRFyoEBS-hledger-flow
        depends base-4.12.0.0
        depends hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        depends optparse-applicative-0.14.3.0-KgUFYVq9ZZSIX6QVzTkb7A
        depends text-1.2.3.1
        depends turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
    definite hledger-flow-0.12.3.1-34mSVQoxqcJ9XqVJgwZ6l9-hledger-flow-test
        depends HUnit-1.6.0.0-LOgQcU2eAMywe67LHWfdn
        depends base-4.12.0.0
        depends containers-0.6.0.1
        depends foldl-1.4.5-KPoJh7Om1qxInpIXJRvbSG
        depends hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
        depends stm-2.5.0.0
        depends text-1.2.3.1
        depends turtle-1.5.14-5ZKr9luCEKQEKdHVVuYVpX
Using Cabal-2.4.0.1 compiled by ghc-8.6
Using compiler: ghc-8.6.5
Using install prefix:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1
Executables installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/bin
Libraries installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/lib/ghc-8.6.5/x86_64-linux-ghc-8.6.5/hledger-flow-0.12.3.1-BxeZsIjrIZtHWzDpRuQVSM
Dynamic Libraries installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/lib/ghc-8.6.5/x86_64-linux-ghc-8.6.5
Private executables installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/libexec/x86_64-linux-ghc-8.6.5/hledger-flow-0.12.3.1
Data files installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/share/x86_64-linux-ghc-8.6.5/hledger-flow-0.12.3.1
Documentation installed in:
/nix/store/mzjyh9m950h1pr3wwvxn2znrdjri87a0-hledger-flow-0.12.3.1-doc/share/doc/hledger-flow-0.12.3.1
Configuration files installed in:
/nix/store/ad3xd6j57q7qjrq4ja7izjixg00a5frr-hledger-flow-0.12.3.1/etc
No alex found
Using ar found on system at:
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 7.4.0 given by user at:
/nix/store/382vgzzm7xrkqfply7y1h3fn03hp68m5-gcc-wrapper-7.4.0/bin/gcc
Using ghc version 8.6.5 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/ghc
Using ghc-pkg version 8.6.5 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.22.0 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/hpc
Using hsc2hs version 0.68.5 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/3x9rflrinbi8pn95myvj25xnmi55qrbp-hscolour-1.24.4/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/5vp528r9xjaj9vdnklggb7366qaywlr3-binutils-wrapper-2.31.1/bin/ld.gold
No pkg-config found
Using runghc version 8.6.5 found on system at:
/nix/store/1acad3xzraihikw74811w0v676z065xm-ghc-8.6.5/bin/runghc
Using strip version 2.31 found on system at:
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/strip
Using tar found on system at:
/nix/store/vggrnz2yn4bcpl0ldrfixdnw2939fdbf-gnutar-1.32/bin/tar
No uhc found
building
Preprocessing library for hledger-flow-0.12.3.1..
Building library for hledger-flow-0.12.3.1..
[1 of 7] Compiling Hledger.Flow.Import.Types ( src/Hledger/Flow/Import/Types.hs, dist/build/Hledger/Flow/Import/Types.o )
[2 of 7] Compiling Hledger.Flow.Types ( src/Hledger/Flow/Types.hs, dist/build/Hledger/Flow/Types.o )
[3 of 7] Compiling Hledger.Flow.RuntimeOptions ( src/Hledger/Flow/RuntimeOptions.hs, dist/build/Hledger/Flow/RuntimeOptions.o )
[4 of 7] Compiling Paths_hledger_flow ( dist/build/autogen/Paths_hledger_flow.hs, dist/build/Paths_hledger_flow.o )
[5 of 7] Compiling Hledger.Flow.Common ( src/Hledger/Flow/Common.hs, dist/build/Hledger/Flow/Common.o )
[6 of 7] Compiling Hledger.Flow.Reports ( src/Hledger/Flow/Reports.hs, dist/build/Hledger/Flow/Reports.o )
[7 of 7] Compiling Hledger.Flow.CSVImport ( src/Hledger/Flow/CSVImport.hs, dist/build/Hledger/Flow/CSVImport.o )
[1 of 7] Compiling Hledger.Flow.Import.Types ( src/Hledger/Flow/Import/Types.hs, dist/build/Hledger/Flow/Import/Types.p_o )
[2 of 7] Compiling Hledger.Flow.Types ( src/Hledger/Flow/Types.hs, dist/build/Hledger/Flow/Types.p_o )
[3 of 7] Compiling Hledger.Flow.RuntimeOptions ( src/Hledger/Flow/RuntimeOptions.hs, dist/build/Hledger/Flow/RuntimeOptions.p_o )
[4 of 7] Compiling Paths_hledger_flow ( dist/build/autogen/Paths_hledger_flow.hs, dist/build/Paths_hledger_flow.p_o )
[5 of 7] Compiling Hledger.Flow.Common ( src/Hledger/Flow/Common.hs, dist/build/Hledger/Flow/Common.p_o )
[6 of 7] Compiling Hledger.Flow.Reports ( src/Hledger/Flow/Reports.hs, dist/build/Hledger/Flow/Reports.p_o )
[7 of 7] Compiling Hledger.Flow.CSVImport ( src/Hledger/Flow/CSVImport.hs, dist/build/Hledger/Flow/CSVImport.p_o )
Preprocessing executable 'hledger-flow' for hledger-flow-0.12.3.1..
Building executable 'hledger-flow' for hledger-flow-0.12.3.1..
[1 of 2] Compiling Main             ( app/Main.hs, dist/build/hledger-flow/hledger-flow-tmp/Main.o )
[2 of 2] Compiling Paths_hledger_flow ( dist/build/hledger-flow/autogen/Paths_hledger_flow.hs, dist/build/hledger-flow/hledger-flow-tmp/Paths_hledger_flow.o )
Linking dist/build/hledger-flow/hledger-flow ...
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lm
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lrt
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lutil
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -ldl
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lpthread
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lgmp
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lm
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lrt
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -ldl
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lffi
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lpthread
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lpthread
/nix/store/m5wwsdqmb62xzjxa7q70x7j46f5b2d1m-binutils-2.31.1/bin/ld: cannot find -lc
collect2: error: ld returned 1 exit status
`cc' failed in phase `Linker'. (Exit code: 1)
builder for '/nix/store/yv96s715lppv7x50a9wkn64x4arp04n2-hledger-flow-0.12.3.1.drv' failed with exit code 1
error: build of '/nix/store/yv96s715lppv7x50a9wkn64x4arp04n2-hledger-flow-0.12.3.1.drv' failed
```

I tried adding `librarySystemDepends = [ stdenv.cc.libc ];` to the `hackage-packages.nix`, but it didn't seem to have an affect on the error message.

--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -1403,4 +1403,11 @@ self: super: {
     HsYAML = self.HsYAML_0_2_1_0;
   };
 
+  # it wants to build a statically linked binary by default
+  hledger-flow = overrideCabal super.hledger-flow ( drv: {
+    postPatch = (drv.postPatch or "") + ''
+      substituteInPlace hledger-flow.cabal --replace "-static" ""
+    '';
+  });
+
 } // import ./configuration-tensorflow.nix {inherit pkgs haskellLib;} self super

--- a/pkgs/development/haskell-modules/configuration-hackage2nix.yaml
+++ b/pkgs/development/haskell-modules/configuration-hackage2nix.yaml
@@ -5874,7 +5874,6 @@ broken-packages:
   - HLearn-distributions
   - hledger-api
   - hledger-chart
-  - hledger-flow
   - hledger-irr
   - hledger-vty
   - hlibBladeRF

